backend/django/create-model-and-api.mo.md

```diff
@@ -0,0 +1,219 @@
+# [MO] Create a model and an api route (~20 mins)
+
+## Owner: Alice Breton
+
+## Prerequisites (~2 mins)
+
+- Have a django project installed, [see this](https://github.com/bamlab/dev-standards/blob/master/backend/django/getting-started.mo.md).
+- Have an app (in this example Publications).
+- In this example we will use an already existing model 'city' in a 'locations' app, to show how to integrate other models with foreign keys.
+
+## Steps  
+
+- In this tutorial we will create a News model, add it to the Django admin back-office and an API route to get all the pieces of news that have been created.
+- To give a full example, we will ==== REMPLIR
+
+### Create a News model *(~10 min)*
+
+- In this example we will create a new model called News, with different attributes as you will see bellow. In the models.py file of your app (in our case Publications) create a new model. 
+
+ourproject/publications/models.py
+```python
+from django.db import models
+
+class News(models.Model):
+    title = models.CharField(max_length=50)   #CharField is a single line form in the admin; max_length speaks for itself
+    description = models.TextField(max_length=1000)   #TextField is a scrollable paragraph in the admin
+    created_at = models.DateTimeField(auto_now_add=True)    #auto_now_add generates automatically a timestamp when you create your create the news
+    updated_at = models.DateTimeField(auto_now=True)    #auto_now updates the timestamp everytime you change the news
+    city = models.ForeignKey('locations.city')    #All the cities of the City model of the locations app
+
+    class Meta:
+        verbose_name_plural = "News"    #The name of the class should be singular, this forces the plural to be News and not Newss
+
+    def __str__(self):
+        return self.title
+```
+
+
+- If you have users that are in groups, they need the permission to access to the Model. We won't explain how to create a group, we just want to show you how to add permissions to them. They will be added automaticcaly when you deploy.
```

Comment:
In Django, Users can be Admin but without any access right to any model.
For instance we would like a Admin User to manage the News, but not the City (reserved to SuperAdmin).
The practice is to create Groups. A Group is a gathering of access rights, a User can belong to a Group hence having the same access rights.
In our case, we want a Group of Mayor to access the news (by default a group has access to nothing).
You could manage Groups directly in the admin interface, but access rights are part of your business logic and business logic should be version controlled in the code.